core/Cargo.toml

```diff
@@ -293,6 +294,8 @@ rusqlite = { version = "0.29.0", optional = true, features = ["bundled"] }
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 sha2 = { version = "0.10", optional = true }
+hmac = "0.12.1"
```

Comment:
Please mark those deps as optional and only enable them when `upyun` feature has been enabled.

Reply:
fixed